docs/01-basics/01-basic-window.md

```diff
@@ -0,0 +1,286 @@
+
+Again declare a target, in this case `01-01-BasicWindow` (I try to keep the docs pages in sync with the target names to aid my brain while writing this).
+Specify which cpp file/unit to build.
+
+Then there is a small block which detects whether you are using Microsoft's C++ Compiler or not, and
+configure the compiler to always treat warnings as errors.
+
+Then we make `spdlog`'s include directory available to the target and link the program with our dependencies.
+
+Finally we can start working on the first c++ parts. We will keep everything in one file, to create a window and react to keyboard input `ESC` and `F11`.
+
+Let's start.
+
+### Main.cpp
+
+I like to use `int32_t`, `uint16_t` over types/aliases like `int`, or `ushort`. For that we include `cstdint`.
+
+We also include glad and glfw header as well as spdlog's. When it comes to glad and glfw we need to be careful here to include glad before glfw.
+C++ is weird and sometimes the order of includes is important, because certain compiler definitions/switches are declared somewhere else which are picked up by something else.
+
+That's one of the cases here.
+
+Plus the usual entry point for your bog standard c/c++ program.
+
+It all should look like this:
+
+```cpp
+#include <cstdint>
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+
+#include <spdlog/spdlog.h>
+
+int main(int32_t argc, char* argv[])
+{
+    return 0;   
+}
+```
+
+You might want to reconfigure the project here, to be sure all deps are loaded and `CMake` knows how to build all this stuff.
+in `VSCode` press `Ctrl+Shift+p` and look for `CMake: Delete Cache and Reconfigure`. It is probably called very similar in `Clion` or `Visual Studio`.
+
+`VSCode` will probably say something like this afterwards:
+
+```bash
+[cmake] -- Build files have been written to: /home/deccer/Projects/OpenGL-Getting-Started/build
+```
+
+If not you most likely goofed something up, please check if you have setup the project structure as I described or if there is a typo somewhere.
+
+If you  try to build it as is. You should get an error saying something like that:
+
+```
+error: unused parameter ‘argc’ [-Werror=unused-parameter]
+```
+
+And another one for `argv` because we dont really use them anywhere. We still want to treat warnings as errors, but we cant really fix anything here,
+instead we tell the compiler that we are aware of it, and annotate the 2 parameters with an attribute called `[[maybe_unused]]`.
+
+The whole thing should look like
+
+```cpp
+#include <cstdint>
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+
+#include <spdlog/spdlog.h>
+
+int main(
+    [[maybe_unused]] int32_t argc,
+    [[maybe_unused]] char* argv[])
+{
+    return 0;   
+}
+```
+
+If you build now, it should succeed.
+
+Next step, create a window. In order to create a window we have to do a few things first, because we want a window which provides us with a render context. To tell `GLFW` what parameters we want we have to give it a few hints, literally.
```

Comment:
~~Next step, create a window. In order to create a window we have to do a few things first, because we want a window which provides us with a render context. To tell `GLFW` what parameters we want we have to give it a few hints, literally.~~
Next step is to create a window with a opengl context which we can use to draw our world into. `GLFW` will handle all that for us, but we need to give it a few hints for setting up the window and context.